drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -110,45 +273,80 @@ void at86rf2xx_reset(at86rf2xx_t *dev)
     at86rf2xx_set_addr_short(dev, &addr_long.uint16[ARRAY_SIZE(addr_long.uint16) - 1]);
 
     /* set default channel */
-    at86rf2xx_set_chan(dev, AT86RF2XX_DEFAULT_CHANNEL);
+    at86rf2xx_set_chan(dev, at86rf2xx_default_channels[dev->base.dev_type]);
     /* set default TX power */
     at86rf2xx_set_txpower(dev, AT86RF2XX_DEFAULT_TXPOWER);
     /* set default options */
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_AUTOACK, true);
     at86rf2xx_set_option(dev, AT86RF2XX_OPT_CSMA, true);
 
     static const netopt_enable_t enable = NETOPT_ENABLE;
-    netdev_ieee802154_set(&dev->netdev, NETOPT_ACK_REQ,
+    netdev_ieee802154_set(&dev->base.netdev, NETOPT_ACK_REQ,
                           &enable, sizeof(enable));
 
     /* enable safe mode (protect RX FIFO until reading data starts) */
     at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_CTRL_2,
                         AT86RF2XX_TRX_CTRL_2_MASK__RX_SAFE_MODE);
-#ifdef MODULE_AT86RF212B
-    at86rf2xx_set_page(dev, AT86RF2XX_DEFAULT_PAGE);
+
+    switch (dev->base.dev_type) {
+#if IS_USED(MODULE_AT86RF212B)
+        case AT86RF2XX_DEV_TYPE_AT86RF212B: {
+            at86rf2xx_set_page(dev, AT86RF212B_DEFAULT_PAGE);
+            break;
+        }
 #endif
+    }
```

Comment:
How about adding a pseudomodule `at86rf2xx_multidev` that is automatically used in case more than a single type of the AT86RF2xx family is used. That would allow to optimize for the case only a single kind of device is used, e.g. like this:

```C
    switch (dev->base.dev_type) {
        default:
#if IS_USED(MODULE_AT86RF2XX_MULTIDEV)
            break;
#endif
#if IS_USED(MODULE_AT86RF212B)
        case AT86RF2XX_DEV_TYPE_AT86RF212B: {
            at86rf2xx_set_page(dev, AT86RF212B_DEFAULT_PAGE);
            break;
        }
#endif
    }
```

With the above code when only `at86rf212b` is used and no other type, the compiler knows for sure that it will just always have to call `at86rf2xx_set_page()` and no checks are needed.

Reply:
To me the issue here is that only Sub GHz radio have different page settings. Do we need a generic `at86rf2xx_set_page`?
As proposed in some other comments, I would define the common code for setting channel page and then only declare variants that make sense (`at86rf231_set_page` should not exist).

Reply:
we would get the same compiler optimization if we only use 2.4 ghz radios.